openeo_fastapi/client/collections.py

```diff
@@ -1,69 +1,87 @@
+from typing import List
+
 import aiohttp
-from fastapi import APIRouter
 
-from openeo_fastapi.client.models import Collection, Collections
-from openeo_fastapi.client.settings import app_settings
+from openeo_fastapi.client.models import Collection, Collections, Endpoint
+from openeo_fastapi.client.register import EndpointRegister
+from openeo_fastapi.client.settings import AppSettings
 
-router_collections = APIRouter()
 
+class CollectionRegister(EndpointRegister):
+    def __init__(self, settings) -> None:
+        super().__init__()
+        self.endpoints = self._initialize_endpoints()
+        self.settings: AppSettings = settings
+        pass
 
-async def get_collections():
-    """
-    Returns Basic metadata for all datasets
-    """
-    stac_url = (
-        app_settings.STAC_API_URL
-        if app_settings.STAC_API_URL.endswith("/")
-        else app_settings.STAC_API_URL + "/"
-    )
+    def _initialize_endpoints(self) -> list[Endpoint]:
+        return [
+            Endpoint(
+                path="/collections",
```

Comment:
do we need `/{self.client.settings.OPENEO_VERSION}/` in the path here as well?

Reply:
Good shout, but I don't think here we want to add it. This is just about setting what we advertise in /capabilities, and we don't actually want the version number to be present when displayed as an output there. Additionally, I initially thought the same when doing the openeo-api-python and the openeo-clients couldn't discover the endpoints xD